Source/WTF/wtf/Logger.h

```diff
@@ -240,14 +241,26 @@ class Logger : public ThreadSafeRefCounted<Logger> {
     template<typename... Arguments>
     inline void debugVerbose(WTFLogChannel& channel, const char* file, const char* function, int line, const Arguments&... arguments) const
     {
-        if (!willLog(channel, WTFLogLevel::Debug))
+        if (!willLog(channel, WTFLogLevel::Debug, arguments...))
             return;
 
         logVerbose(channel, WTFLogLevel::Debug, file, function, line, arguments...);
     }
 
-    inline bool willLog(const WTFLogChannel& channel, WTFLogLevel level) const
+    template<typename... Argument>
+    inline bool willLog(const WTFLogChannel& channel, WTFLogLevel level, const Argument&... arguments) const
     {
+        {
+            if (!observerLock().tryLock())
+                return false;
+
+            Locker locker { AdoptLock, observerLock() };
+            for (Observer& observer : observers()) {
+                if (observer.handleLogMessage(channel, level, { ConsoleLogValue<Argument>::toValue(arguments)... }))
```

Comment:
Why not doing the work in `didLogMessage` instead?

Reply:
Because then this would become dependent on having release logs enabled at runtime (on Linux ports by setting this env var: WEBKIT_DEBUG=WebRTC=enabled) and thus the logs would also be sent to systemd journal.

The `didLogMessage` handler is not called if nothing was logged, that's why I added a new method.

Reply:
OK. Do we need `handleLogMessage` to return a boolean?
It seems it could only be a passive listener.